File: awa/sdk_dist/python/awa/client/utils/general/get_workflow_paths.py
from pathlib import Path

from temporalio.workflow import Info

from awa.client.models.workflow_paths import WorkflowPaths
from awa.client.utils.general.get_workflow_paths_direct import get_workflow_paths_direct


def get_workflow_paths(workflow_dir: Path, workflow_info: Info) -> WorkflowPaths:
    """Get workflow paths using Temporal workflow info.

    Args:
        workflow_dir: The base workflow directory.
        workflow_info: Temporal workflow info containing workflow type and ID.

    Returns:
        WorkflowPaths: Object containing all relevant workflow paths.

    """
    return get_workflow_paths_direct(
        workflow_dir,
        workflow_info.workflow_type,
        workflow_info.workflow_id,
    )

File: awa/sdk_dist/csharp/Awa.Client/Utilities/General/GetWorkflowPaths.cs
using System;
using Awa.Client.Models;

namespace Awa.Client.Utilities.General
{
    /// <summary>
    /// General utility functions for AWA SDK.
    /// </summary>
    public static partial class GeneralUtilities
    {
        /// <summary>
        /// Get workflow paths using Temporal workflow info.
        /// </summary>
        /// <param name="workflowDir">The base workflow directory.</param>
        /// <param name="workflowType">Temporal workflow type.</param>
        /// <param name="workflowId">Temporal workflow ID.</param>
        /// <returns>WorkflowPaths object containing all relevant workflow paths.</returns>
        public static WorkflowPaths GetWorkflowPaths(string workflowDir, string workflowType, string workflowId)
        {
            return GetWorkflowPathsDirect(workflowDir, workflowType, workflowId);
        }
    }
}
